parser/src/lexer/cursor.rs

```diff
@@ -0,0 +1,108 @@
+use crate::text_size::{TextLen, TextSize};
+use std::str::Chars;
+
+pub(crate) const EOF_CHAR: char = '\0';
+
+#[derive(Clone, Debug)]
+pub(super) struct Cursor<'a> {
```

Comment:
Are there many changes here compared to the version in `ruff_python_trivia`?

Reply:
Nope. I used this as template for the Cursor in `ruff_python_trivia`. It has two functional differences:

* It has a method to get the rest of the string
* It has a method that gives you the previous token in debug builds